src/Microsoft.OData.Core/Evaluation/ODataConventionalIdMetadataBuilder.cs

```diff
@@ -131,11 +131,11 @@ private void ComputeAndCacheId()
                 case EdmNavigationSourceKind.Singleton:
                     uri = this.ComputeIdForSingleton();
                     break;
+                // Treat UnknownEntitySet as a containment
+                case EdmNavigationSourceKind.UnknownEntitySet:
                 case EdmNavigationSourceKind.ContainedEntitySet:
                     uri = this.ComputeIdForContainment();
                     break;
-                case EdmNavigationSourceKind.UnknownEntitySet:

```

Comment:
Is it a behavior breaking change?

Reply:
It is turning an error case into a success case.